omod/src/main/java/org/openmrs/web/controller/PortletController.java

```diff
@@ -399,30 +479,56 @@ public ModelAndView handleRequest(HttpServletRequest request, HttpServletRespons
 						Concept c = cs.getConcept(i);
 						concepts.put(i, c);
 						conceptsByStringIds.put(i.toString(), c);
-					}
-					catch (Exception ex) {
+					} catch (Exception ex) {
 						log.error("Error during putting int i into concept c", ex);
 					}
 				}
-				
+
 				model.put("conceptMap", concepts);
 				model.put("conceptMapByStringIds", conceptsByStringIds);
 			}
-			
+
 			populateModel(request, model);
 			log.debug(portletPath + " took " + (System.currentTimeMillis() - timeAtStart) + " ms");
 		}
-		
+
 		return new ModelAndView(portletPath, "model", model);
-		
+
 	}
-	
+
 	/**
-	 * Subclasses should override this to put more data into the model. This will be called AFTER
-	 * handleRequest has put mappings in the model as described in its javadoc. Note that context
+	 * Subclasses should override this to put more data into the model. This will be
+	 * called AFTER
+	 * handleRequest has put mappings in the model as described in its javadoc. Note
+	 * that context
 	 * could be null when this method is called.
 	 */
 	protected void populateModel(HttpServletRequest request, Map<String, Object> model) {
 	}
-	
+
+	private Integer getPageParameter(HttpServletRequest request) {
+		try {
+			return Integer.parseInt(request.getParameter("page"));
+		} catch (NumberFormatException e) {
+			return 0;
+		}
+	}
+
+	private Integer getPageSizeParameter(HttpServletRequest request, AdministrationService as) {
+		try {
+			String pageSizeParam = request.getParameter("pageSize");
+			if (pageSizeParam != null) {
+				return Integer.parseInt(pageSizeParam);
+			}
+
+			String globalPageSize = as.getGlobalProperty("dashboard.encounters.maximumNumberToShow");
```

Comment:
Is this pull request about encounters or observations?

Reply:
And just in case you find this helpful: https://openmrs.atlassian.net/browse/LUI-188

Reply:
Yes, we thought we could align encounters as well to limit it to based max numbers to show or default to the fallback property 50. cc @eudson  
Encounters fetched are now limited to the value of the GP

Reply:
Okay so that solution effectively decouples the configuration for the Form Entry tab and the Encounters tab which tackles both usability and configurability needs without disrupting existing any setups. Are you trying to say we employ the same thing but for the patientEncounters jsp, LOL, I am having a brain fart right now with looking at old tech am not as familiar with

Reply:
One should be able to set different values for encounters and observations.

Reply:
This involves creating a different GP for Obs as in `maximumNumberObservationsToShow`? Please advise

Reply:
Yes you need to create a separate global property for it.